### ISSUE TITLE
IDE0018 (Variable declaration can be inlined) invalid fix

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#18668]
**Version Used**: VS 15.1 (26403.3)

Example:
```C#
static void M(bool condition)
{
    var x = 1;
    var result = condition && int.TryParse("2", out x);
    Console.WriteLine(x);
}
```
IDE0018 is detected for variable 'x' (it should not be detected). After the fix code looks like this:
```C#
static void M(bool condition)
{
    var result = condition && int.TryParse("2", out int x);
    Console.WriteLine(x);
}
```
which does not compile because of "CS0165	Use of unassigned local variable 'x'" error (information that 'x' was set to '1' is completely lost).

It's interesting that when you change original code ("var" -> "int"):
```C#
static void M(bool condition)
{
    int x = 1;
    var result = condition && int.TryParse("2", out x);
    Console.WriteLine(x);
}
```
then the IDE0018 is correctly not detected at all.
